docs/partitioning.md

```diff
@@ -1,52 +1,150 @@
+table into several smaller tables, each containing variants of a specific region of a
+genome. For example, you can have one output table per chromosome, in that
+case the above query can be written as:
+
+```
+SELECT * from `1000genomes_chr17`
+WHERE start_position >= 41197694
+  AND end_position <= 41276113
+```
+
+Note that condition on the `reference_name` is removed since we know this table,
+as its name suggests, only contains the variants of 17th chromosome.
+
+By splitting output tables based on the `reference_name` you are
+guaranteed that per-chromosome queries will only process variants of the
+chromosome under study. Also, appending new rows to existing tables does not
+impact this guarantee on the cost, unlike the BigQuery clustering solution.
+
+This solution has some limitations comparing to clustering. For example, you
+will be charged for processing of a whole chromosome's table even if only a
+small region is being processed. As an example, the previous query, will cost
+152 GB or $0.74. This is significantly less than the original cost without
+partitioning but it's more than clustering cost.
+
+You could define your partitions to be more fine grained and have multiple
+tables per chromosome. However, you need to anticipate how your future queries
+are going to be in order to optimize your output partitions. Since in many
+use cases it not obvious to anticipate future queries, we offer the
+third solution as the most practical and cost effective solution.
+
+## Solution 3: Hybrid Solution
+
+This solution combines two previous solutions to offer the benefits of both.
+Using the Variant Transforms' native partitioning output table will be split
+into several smaller tables (perhaps one table per chromosome) and then each
+table will be clustered based on the `start_position, end_position` columns to
+further optimize them for running queries.
+
+Using this technique the *BRCA1 query* will cost 226 MB or $0.0011 which is more
+than half of the clustering cost. In our experimental studies, we found out the
+most effective way to reduce query cost, specially *the point lookup queries*,
+is this hybrid solution.
+
+If you append new rows to your clustered table and your table gradually becomes
+partially sorted, you still have a strict guarantee that your query cost will
+be limited to the size of the partitioned table ($0.74 in this case).
+Also, partitioning output table into several smaller tables reduces the initial
+clustering time significantly.
+
+In the following section we will explain how you could use Variant Transforms
+to easily partition your output table to minimize the cost of your future
+queries.
 
 ## Partition Config files
 
-Output table is partitioned based on the *partition config file* which can be
-set using `--partition_config_path` flag. The config file is formatted as
+Variant Transforms output table is partitioned based on the *partition config
```

Comment:
Consider rephrasing the initial sentence to be more explicit about the solutions...e.g. `Solution #2 or #3 require a partition config file to define the output tables. The file is specified using the --partition_config_path flag and is formatted as ...`

Reply:
done.